src/app/src/App.js

```diff
@@ -43,12 +43,22 @@ class App extends Component {
             containerClassName += ' modal-is-open';
         }
 
+        const sensorReadingDates = sensors.features.map(sensor => {
+            const sensorId = sensor.properties.Id;
+            return sensorData[sensorId].timestamp;
+        });
+        const mostRecentDate = Math.max(...sensorReadingDates);
+        const msElapsedSinceLastUpdate =
+            mostRecentDate === 0 ? null : new Date().getTime() - mostRecentDate;
```

Comment:
Consider moving most of this to another utility function and adding a test (thinking of it as a function that extracts the most recent date from a list). This also may be better placed directly in the `Footer` component.

Reply:
I thought about this through it's inception. It's kind of weird no matter what we send through to the footer. Sensors and Sensor data? Too much information. A list dates? Maybe but feels a little intermediary. Most recent date? Maybe. Time since last update? Maybe.

I'll see if turning some of this into utils and writing a test helps at all with organization.

Reply:
Changed as suggested. The placement of the logic in `Footer` is maybe a little better, so I'll take it. I added an additional test, but I don't think it's fundamentally a good test. It tests that the spread operator and `Math.max`operator work, not any custom logic.

Reply:
Cool, yeah maybe that was overkill, though I'm becoming more and more putting as much of the data work in utility functions that we can test, and leaving the components more simple.